masonry/src/render_root.rs

```diff
@@ -576,7 +576,11 @@ impl RenderRoot {
         // We request a redraw if either the render tree or the accessibility
         // tree needs to be rebuilt. Usually both happen at the same time.
         // A redraw will trigger a rebuild of the accessibility tree.
-        if self.root_state().needs_paint || self.root_state().needs_accessibility {
+        // TODO - We assume that a relayout will trigger a repaint
+        if self.root_state().needs_paint
+            || self.root_state().needs_accessibility
+            || self.root_state().needs_layout
```

Comment:
Quick comment on this: right now layout is implicitly triggered as part of the paint/accessibility pass: you want to paint something, therefore you lay it out first.

A side-effect of this is that previous to this change, calling request_layout wouldn't call a layout pass until *some* widget somewhere requested a paint pass. (Even though the layout pass automatically requests a paint pass.)

Once pass spec is complete, layout will run with all the other passes whenever anything happens, and this line will be removed.